### PR TITLE
Update local testing workflow instructions

### LINKED ISSUE #129: missing artifact download link/testing/instruction/script for local testing.

### PR DESCRIPTION
**Pull request by waltsims**
close #129

### CODE REVIEW

**Comment by waltsims** [comment]
In the process of making this PR, I removed unused tests that are not run in the CI and added optional dependency installation testing in the CI.

File: .github/workflows/test_optional_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            - uses: actions/checkout@v4
+            # Pull the cache based on the hash value
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v5
+              with:
+                  python-version: ${{ matrix.python-version }}
+                  cache: 'pip'
+            - name: Install dependencies
+              run: |
+                  pip install '.[${{ matrix.extra_requirements }}]'
```
Comment (waltsims) [inline .github/workflows/test_optional_dependencies.yml:23]: Perhaps we can reduce the number of runs by just running "all" for all configurations.

File: docs/development/development_environment.rst [hunk 2]
```diff
@@ -2,31 +2,48 @@ Development Environment
...
 
+- **Integration testing**: k-Wave-python tests output .h5 files that are passed to the k-Wave binaries and ensures that they match the output of the original k-Wave.
+  This testing compares the output for many of the example scripts from the original k-Wave package.
+  Hash values of the reference output ``.h5`` file from MATLAB examples are generated and stored in ``.json`` files in ``tests/reference_outputs/``.
+  This ``.json`` files are contained in the code repository and do not need to be generated.
+  Since these files are generated from the original k-Wave package, they only need to be updated when a new release of k-Wave is made.
 
-Test References
-=======================
+<TODO: @farid How to regenerate theses outputs?>
```
Comment (waltsims) [inline docs/development/development_environment.rst:37]: @faridyagubbayli Your general feedback is welcome, but please describe how to generate the .json files here.

**Review by faridyagubbayli — APPROVED** [review]
Some suggestions that are not blockers.

**Comment by faridyagubbayli** [comment]
Tests fail because `pre-commit 3.6.2` is not available for Python 3.8. @waltsims do you think we can downgrade? Alternatively, we have another PR to drop Python 3.8 support: https://github.com/waltsims/k-wave-python/pull/246

**Comment by waltsims** [comment]
Yes

**Comment by faridyagubbayli** [comment]
As python 3.8 support was dropped, I removed it from the pipeline's test list and now tests are passing.

File: docs/development/development_environment.rst [hunk 3]
```diff
@@ -2,31 +2,67 @@ Development Environment
...
-With this, you are ready to develop k-Wave-python.
-If you have any issues or questions, please post them on the `k-Wave-python discussions page <https://github.com/waltsims/k-wave-python/discussions>`_ to discuss. We look forward to interacting with you.
+- **Unit testing**: k-Wave-python functions that have a direct counterpart in original k-Wave are tested by comparing the outputs of the two functions.
+  The output of the original k-Wave functions are stored in ``.mat`` files.
+  These files can be generated by running the corresponding MATLAB scripts located in the ``tests/matlab_test_data_collectors/matlab_collectors/`` directory by running ``tests/matlab_test_data_collectors/run_all_collectors.m``.
+  After running the scripts, the reference files can be found in ``tests/matlab_test_data_collectors/python_testes/collectedValues/``.
+ 
+.. note::
+    If you do not have MATLAB installed to generate the reference files, you can download recently generated reference file outputs from the GitHub CI and place them in the ``python_testers/collectedValues/`` directory.
+    The latest reference files can be found in the artifacts of the latest CI run of ``pytest.yml``.
```
Comment (waltsims) [inline docs/development/development_environment.rst:29]:
```suggestion
    The latest reference files can be found in the artifacts of the latest CI run of ``pytest.yml`` (e.g. [here](https://github.com/waltsims/k-wave-python/actions/runs/7770639710/artifacts/1217868112)).
```

**Pull request merged**